### PR TITLE
Add DigitalOcean module

### PR DESCRIPTION
**Pull request by pothos**
The setup is the same as in, e.g., Hetzner.

### CODE REVIEW

**Review by krnowak — COMMENTED** [review]
I think you should mention the new `digitalocean` directory in the toplevel README too.

File: digitalocean/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+```
+export DIGITALOCEAN_TOKEN=...
+terraform init
+terraform apply
+```
+
+Log in via `ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@IPADDRESS`.
+
+When you make a change to `machine-mynode.yaml.tmpl` and run `terraform apply` again, the instance will be replaced. Consider to use [`create_before_destroy`](https://www.terraform.io/docs/configuration/meta-arguments/lifecycle.html#syntax-and-arguments) in your final setup.
```
Comment (marga-kinvolk) [inline digitalocean/README.md:24]:
```suggestion
When you make a change to `machine-mynode.yaml.tmpl` and run `terraform apply` again, the instance will be replaced. Consider using [`create_before_destroy`](https://www.terraform.io/docs/configuration/meta-arguments/lifecycle.html#syntax-and-arguments) in your final setup.
```

File: digitalocean/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Now run Terraform (version 13) as follows:
+
+```
+export DIGITALOCEAN_TOKEN=...
+terraform init
+terraform apply
+```
+
+Log in via `ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null core@IPADDRESS`.
```
Comment (marga-kinvolk) [inline digitalocean/README.md:22]: For non-ephemeral instances, this seems like a bad practice.

File: digitalocean/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# DigitalOcean
+
+This small setup can be used to provision Flatcar nodes on [DigitalOcean](https://www.digitalocean.com/).
+
+Edit the [Container Linux Config](https://kinvolk.io/docs/flatcar-container-linux/latest/container-linux-config-transpiler/configuration/) `machine-mynode.yaml.tmpl` file if you like, then create the following `terraform.tfvars` with a machine `mynode`, corresponding to the Container Linux Config file name. If you add more machines, create new files for them.
```
Comment (marga-kinvolk) [inline digitalocean/README.md:5]: I find this instructions a bit confusing. If I'm understanding correctly, if I want to have a machine called "marga" then I need to copy machine-mynode.yaml.tmpl and name it machine-marga.yaml.tmpl, and then use "marga" in the machines list, is that correct?

If so, I think I'd word this the other way round:

Create a `terraform.tfvars` file that lists your preferences. Like this one:
```
cluster_name = "mycluster"
machines     = ["mynode"]
datacenter   = "nyc3"
ssh_keys     = ["ssh-rsa AA... me@mail.net"]
```

The machine name listed in the `machines` variable is used to retrieve the corresponding [Container Linux Config](https://kinvolk.io/docs/flatcar-container-linux/latest/container-linux-config-transpiler/configuration/). For each new machine you add to the list, you should copy and modify `machine-mynode.yaml.tmpl` to the corresponding name.

File: digitalocean/digitaloecan-droplets.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  }
+}
+
+resource "digitalocean_ssh_key" "first" {
+  name       = var.cluster_name
+  public_key = var.ssh_keys.0
+}
+
+resource "digitalocean_custom_image" "flatcar" {
+  name   = "flatcar-stable-2605.12.0"
```
Comment (marga-kinvolk) [inline digitalocean/digitaloecan-droplets.tf:29]: Same comment as in the docs, the version number should be a variable.

File: digitalocean/machine-mynode.yaml.tmpl [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  files:
+    - path: /home/core/works
+      filesystem: root
+      mode: 0755
+      contents:
+        inline: |
+          #!/bin/bash
+          set -euo pipefail
+          hostname="$(hostname)"
+          echo My name is ${name} and the hostname is $${hostname}
```
Comment (marga-kinvolk) [inline digitalocean/machine-mynode.yaml.tmpl:16]: What's the point of this script?  How does "${name}" get filled?

Comment (pothos) [reply digitalocean/machine-mynode.yaml.tmpl:16]: The script demonstrates using the Terraform template language with Terraform variables and shell variables, `${name}` is the Terraform substitution from
```
  vars = {
   ssh_keys = jsonencode(var.ssh_keys)
    name     = each.key
  }
```
while `$${hostname}` is the quoted `${hostname}` – I think this is the main thing to be aware of for when using Container Linux Configs with Terraform, so it's good to have this covered in the example.

Comment (marga-kinvolk) [reply digitalocean/machine-mynode.yaml.tmpl:16]: I see, please add a comment about this, then:
# This script demonstrates how templating and variable substitution works when using Container Linux Configs.

File: digitalocean/README.md (hunk 2 above)
Comment (pothos) [reply digitalocean/README.md:22]: For immutable infra where the whole instance is replaced it somehow makes sense, but I can turn this into a suggesstion.

File: digitalocean/README.md (hunk 1 above)
Comment (pothos) [reply digitalocean/README.md]: Done

File: digitalocean/README.md (hunk 3 above)
Comment (pothos) [reply digitalocean/README.md]: Thanks, yes, this order is better to follow. Made a few changes to clarify the relation of the list with the files.

File: digitalocean/digitaloecan-droplets.tf (hunk 4 above)
Comment (pothos) [reply digitalocean/digitaloecan-droplets.tf]: Done

File: digitalocean/machine-mynode.yaml.tmpl (hunk 5 above)
Comment (pothos) [reply digitalocean/machine-mynode.yaml.tmpl:17]: Thanks, done

**Review by marga-kinvolk — APPROVED** [review]
LGTM, just a typo and a suggestion for an extra sentence.

**Pull request merged**